### PR TITLE
Update to terraform 0.12.6

### PR DESCRIPTION
**Pull request by akatrevorjay**

### CODE REVIEW

**Review by arwalls — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,64 +3,130 @@ version: 2
...
+          name: checkout-submodules
           command: |
-            apk add --no-cache \
-              bash \
-              curl \
-              docker \
-              jq
+            set -eo pipefail
+
+            git submodule sync
```
Comment (hinling-whistle) [inline .circleci/config.yml:26]: question: what submodules are we syncing?

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,64 +3,130 @@ version: 2
 jobs:
   build:
     docker:
-      - image: alpine:3.7
+      - image: circleci/python:latest
```
Comment (hinling-whistle) [inline .circleci/config.yml:6]: Do we need python as the base? or ruby?

File: .circleci/config.yml [hunk 3]
```diff
@@ -3,64 +3,130 @@ version: 2
...
-            mkdir -p ~/docker
-            docker save "${CI_IMAGE_NAME}":latest > ~/docker/image.tar
+            set -eo pipefail
+
+            build_tag="${CI_IMAGE_NAME}:build-${CIRCLE_BUILD_NUM}"
+
+            mkdir -pv ~/docker
+
+            # The history is also required to get each individual layer.
+            docker save "$build_tag" $(docker history -q "$build_tag" | grep -v '<missing>') > ~/docker/image.tar
```
Comment (hinling-whistle) [inline .circleci/config.yml:90]: yea...need the `grep -v '<missing>'`. I did that with the server repo build also.

File: .circleci/config.yml (hunk 1 above)
Comment (akatrevorjay) [reply .circleci/config.yml:26]: None it's just good practice to always do it in CI.

`checkout` should do it, but circle leaves a lot to be desired.

File: .circleci/config.yml (hunk 2 above)
Comment (akatrevorjay) [reply .circleci/config.yml:6]: We need neither, I am just biased, and there is no "minimal" image.

**Pull request merged**